zio-schema/shared/src/main/scala/zio/schema/DynamicValue.scala

```diff
@@ -1866,4 +1871,592 @@ object DynamicValue {
+        Schema.Field("values", Schema.defer(Schema.chunk(DynamicValueSchema()))),
+        chunk => DynamicValue.Sequence(chunk),
+        seq => seq.values
+      ),
+      _.asInstanceOf[DynamicValue.Sequence]
+    )
+
+  private val enumerationCase: Schema.Case[DynamicValue.Enumeration, DynamicValue] =
+    Schema.Case(
+      "Enumeration",
+      Schema.CaseClass1[(String, DynamicValue), DynamicValue.Enumeration](
+        Schema.Field(
+          "value",
+          Schema.defer(Schema.tuple2(Schema.primitive(StandardType.StringType), DynamicValueSchema()))
+        ),
+        value => DynamicValue.Enumeration(value),
+        enumeration => enumeration.value
+      ),
+      _.asInstanceOf[DynamicValue.Enumeration]
+    )
+
+  private val recordCase: Schema.Case[DynamicValue.Record, DynamicValue] =
+    Schema.Case(
+      "Record",
+      Schema.CaseClass1[Map[String, DynamicValue], DynamicValue.Record](
+        Schema.Field(
+          "values",
+          Schema.defer(Schema.map(Schema.primitive(StandardType.StringType), DynamicValueSchema()))
+        ),
+        map => DynamicValue.Record(map.asInstanceOf[ListMap[String, DynamicValue]]),
+        record => record.values
+      ),
+      _.asInstanceOf[DynamicValue.Record]
+    )
+
+  private val dynamicAstCase: Schema.Case[DynamicValue.DynamicAst, DynamicValue] =
+    Schema.Case(
+      "DynamicAst",
+      Schema.CaseClass1[SchemaAst, DynamicValue.DynamicAst](
+        Schema.Field("ast", SchemaAst.schema),
+        schemaAst => DynamicValue.DynamicAst(schemaAst),
+        dynamicAst => dynamicAst.ast
+      ),
+      _.asInstanceOf[DynamicValue.DynamicAst]
+    )
+
+//  private def singletonCase[A]: Schema.Case[DynamicValue.Singleton[A], DynamicValue] = ???
+  // private def singletonCase[A](implicit schema: Schema[A]): Schema.Case[DynamicValue.Singleton[A], DynamicValue] =
+  //   Schema.Case(
+  //     "Singleton",
+  //     Schema.CaseClass1[A, DynamicValue.Singleton[A]](
+  //       Schema.Field("instance", Schema.defer(schema)),
+  //       instance => DynamicValue.Singleton(instance),
+  //       singleton => singleton.instance
+  //     ),
+  //     _.asInstanceOf[DynamicValue.Singleton[A]]
+  //   )
+
+  private val primitiveUnitCase: Schema.Case[DynamicValue.Primitive[Unit], DynamicValue] =
+    Schema.Case(
+      "Primitive",
+      Schema.CaseClass2[Unit, StandardType[Unit], DynamicValue.Primitive[Unit]](
+        Schema.Field("value", Schema.primitive(StandardType.UnitType)),
+        Schema.Field("standardType", unitSchema.transform(_ => StandardType.UnitType, _ => ())),
+        (value, standardType) => DynamicValue.Primitive(value, standardType),
+        primitive => primitive.value,
+        primitive => primitive.standardType
+      ),
+      _.asInstanceOf[DynamicValue.Primitive[Unit]]
+    )
+
+  private val primitiveStringCase: Schema.Case[DynamicValue.Primitive[String], DynamicValue] =
+    Schema.Case(
+      "Primitive",
```

Comment:
We want the case IDs to be unique. So I think here we should use the primitive type name instead of "Primitive." That way we can use the simplest encoding. So I think something like:

```scala
Schema.Case(
  "String",
  Schema[String].transform(s => DynamicValue.Primitive(s, StandardType[String], _.value),
 _.asInstanceOf[DynamicValue.Primitive[String]
)
```

Reply:
Oh I see!
For some reason I thought I need to avoid getting specific schemas implicitly)
This look so much better!
Thank you)